docs/source/reference/api-server/examples/api-server-in-docker.rst

```diff
@@ -0,0 +1,100 @@
+.. _sky-api-server-in-docker:
+
+Deploy SkyPilot API Server in Docker
```

Comment:
The original deployment doc is just too long and complicated -- keep this in a separate doc